src/auth/auth.service.ts

```
@@ -0,0 +1,30 @@
+  async register(registerDto: RegisterDto): Promise<Omit<User, 'password'>> {
+    const { email, password, role } = registerDto;
+  
+    const existing = await this.userRepository.findOne({ where: { email } });
+    if (existing) throw new Error('Email already exists');
+  
+    const hashed = await bcrypt.hash(password, 10);
+  
+    const user = this.userRepository.create({ email, password: hashed, role });
+    const saved = await this.userRepository.save(user);
+  
+    const { password: _, ...safeUser } = saved;
```

ESLint

[error] '_' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)